### PR TITLE
add first draft seaexplorer cdl

### PR DESCRIPTION
**Pull request by callumrollo**
Proponents: 
Moderator: @OceanGlidersCommunity/format-mantainers

# Type of PR

- [ ] Typo without possible change of interpretation of the related text.
- [ ] Fix of some error, inconsistency, unforeseen limitation.
- [ ] Style that only affects visually the compiled document
- [x] Addition that does not require change in the current structure.
- [ ] Enhancement that require changes to improve the format.

# Related Issues

# Dates when it got review approvals

# Release checklist

- [ ] Approved by at least two members of the committee?
- [ ] There were modifications after the review approvals? If so, please
      ask reviewers to update their review.
- [ ] Proponents and moderador should explicitly agree that it is ready to
      to merge.
- [ ] The moderador is the one in charge to actually merge or close this PR
      according to the final decision.

# For maintainers

- [ ] Update the moderator with a volunteer from the committee. It would be
      best to have one single moderator to guide and help this PR to move
      forward. It is OK to update the moderador pass it to another one.
- [ ] Confirm that the associated branch was deleted after the merging.
- [ ] Wrap-up and close the related issues.

# Comments

### CODE REVIEW

**Comment by callumrollo** [comment]
This is a contribution to the /examples directory started by Victor over in #171 

This example uses seaexplorer data and I have tested it in ERDDAP. It behaves as expected as a columnar dataset with N_MEASUREMENTS the only dimension.

There are still a couple of errors when testing against CF1.8. Working on it

**Review by glidermann — APPROVED** [review]
very nice work

**Comment by callumrollo** [comment]
Ready for review @emmerbodc  @vturpin

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+	double TIME_GPS(N_MEASUREMENTS) ;
+		TIME_GPS:_FillValue = NaN ;
+		TIME_GPS:long_name = "TIME of each GPS location" ;
+		TIME_GPS:observation_type = "measured" ;
+		TIME_GPS:standard_name = "time" ;
+		TIME_GPS:URI = "https://vocab.nerc.ac.uk/collection/P02/current/AYMD/" ;
+		TIME_GPS:coordinates = "DEPTH LATITUDE LONGITUDE TIME" ;
+		TIME_GPS:units = "seconds since 1970-01-01T00:00:00+00:00" ;
+		TIME_GPS:calendar = "proleptic_gregorian" ;
+	char TRAJECTORY(string24) ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:230]: The manual defines trajectory as a `string` type, and by doing so we can avoid these `stringxx` dimensions.

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+		CNDC:long_name = "Electrical conductivity of the water body by CTD" ;
+		CNDC:observation_type = "measured" ;
+		CNDC:standard_name = "sea_water_electrical_conductivity" ;
+		CNDC:units = "mS cm-1" ;
+		CNDC:valid_max = 85. ;
+		CNDC:valid_min = 0. ;
+		CNDC:URI = "https://vocab.nerc.ac.uk/collection/OG1/current/CNDC/" ;
+		CNDC:ancillary_variables = "CNDC_QC" ;
+		CNDC:coordinates = "DEPTH LATITUDE LONGITUDE TIME" ;
+	int64 CNDC_QC(N_MEASUREMENTS) ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:20]: It's not much, but you can save in size using `int32` or `int16`.

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+		CNDC_QC:quality_control_set = 1LL ;
+		CNDC_QC:valid_min = 1LL ;
+		CNDC_QC:valid_max = 9LL ;
+		CNDC_QC:flag_values = 1LL, 2LL, 3LL, 4LL, 9LL ;
+		CNDC_QC:flag_meanings = "GOOD UNKNOWN SUSPECT FAIL MISSING" ;
+		CNDC_QC:long_name = "Electrical conductivity of the water body by CTD Quality Flag" ;
+		CNDC_QC:standard_name = "status_flag" ;
+		CNDC_QC:comment = "Quality control flags from IOOS QC QARTOD https://github.com/ioos/ioos_qc Version: 2.1.0. Using config: [<Call stream_id=conductivity function=qartod.gross_range_test(suspect_span=[6, 42], fail_span=[3, 45])>, <Call stream_id=conductivity function=qartod.location_test(bbox=[10, 50, 25, 60])>].  Threshold values from EuroGOOS DATA-MEQ Working Group (2010) Recommendations for in-situ data Near Real Time Quality Control [Version 1.2]. EuroGOOS, 23pp. DOI http://dx.doi.org/10.25607/OBP-214." ;
+		CNDC_QC:coordinates = "DEPTH LATITUDE LONGITUDE TIME" ;
+	float PHASE(N_MEASUREMENTS) ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:32]: Maybe use an integer type here? Probably `int8` would be enough?

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+
+// global attributes:
+		:AD2CP = "{\'calibration_date\': \'2022-03-16\', \'factory_calibrated\': \'Yes\', \'long_name\': \'Nortek Glider1000 AD2CP\', \'make\': \'Nortek\', \'make_model\': \'Nortek AD2CP\', \'model\': \'AD2CP\', \'serial\': \'103356\'}" ;
+		:Conventions = "CF-1.8, OG-1.0" ;
+		:Metadata_Conventions = "CF-1.6, Unidata Dataset Discovery v1.0" ;
+		:acknowledgement = "This study used data collected and made freely available by Voice of the Ocean Foundation (https://voiceoftheocean.org) accessed from https://erddap.observations.voiceoftheocean.org/erddap/index.html" ;
+		:basin = "Western Gotland Basin" ;
+		:cdm_data_type = "Trajectory" ;
+		:comment = "deployment and recovery in Gotland" ;
+		:contributor_name = "Callum Rollo, Louise Biddle, Olle Petersson, Aleksandra Mazur, Marcus Melin, Gunnar Johnsson, Andrew Birkett, Chiara Monforte" ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:258]: Great job adding the full team!

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+		:contributor_role = "Data scientist, PI, Operator, Operator, Operator, Operator, Operator, Operator," ;
+		:creator_email = "callum.rollo@voiceoftheocean.org" ;
+		:creator_name = "Callum Rollo" ;
+		:creator_url = "https://observations.voiceoftheocean.org" ;
+		:ctd = "{\'calibration_date\': \'2021-03-01\', \'factory_calibrated\': \'Yes\', \'long_name\': \'RBR legato CTD\', \'make\': \'RBR\', \'make_model\': \'RBR legato CTD\', \'model\': \'legato\', \'serial\': \'206523\'}" ;
+		:dataset_id = "nrt_SEA076_M19" ;
+		:date_created = "2023-09-15T08:05:17Z" ;
+		:date_issued = "2023-09-15T08:05:17Z" ;
+		:date_modified = " " ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:267]: When it was the very first version, I used to repeat the `date_created`.

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+		string :variables = "time", "latitude", "longitude", "nav_state", "heading", "pitch", "roll", "dive_num", "security_level", "declination", "internal_temperature", "internal_pressure", "desired_heading", "ballast_cmd", "ballast_pos", "linear_cmd", "linear_pos", "angular_cmd", "angular_pos", "voltage", "altimeter", "dead_reckoning", "nav_resource", "conductivity", "temperature", "pressure", "salinity", "chlorophyll", "chlorophyll_raw", "cdom", "cdom_raw", "backscatter_scaled", "backscatter_raw", "oxygen_concentration", "temperature_oxygen", "ad2cp_heading", "ad2cp_pitch", "ad2cp_roll", "ad2cp_pressure", "ad2cp_time", "ad2cp_beam1_cell_number1", "ad2cp_beam2_cell_number1", "ad2cp_beam3_cell_number1", "ad2cp_beam4_cell_number1" ;
+		:wmo_id = "6801665" ;
+		:deployment_start = "2023-09-06T08:52:59.375000000" ;
+		:deployment_end = "2023-09-15T07:37:07.016000000" ;
+		:disclaimer = "Data, products and services from VOTO are provided \'as is\' without any warranty as to fitness for a particular purpose." ;
+		:platform_vocabulary = "https://vocab.nerc.ac.uk/collection/L06/current/27/" ;
+		:contributor_email = "callum.rollo@voiceoftheocean.org, louise.biddle@voiceoftheocean.org" ;
```
Comment (castelao) [inline examples/SEA076_20230906T0852_nrt.cdl:316]: Could `contributor_email` has only two items while `contributor_name` has more? I don't know the answer for that one.

**Review by castelao — COMMENTED** [review]
Great job @callumrollo ! Some suggestions and comments.

File: examples/SEA076_20230906T0852_nrt.cdl [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+		:rtqc_method = "IOOS QC QARTOD https://github.com/ioos/ioos_qc" ;
+		:rtqc_method_doi = "None" ;
+data:
+
+ CNDC = -0.0041, -0.0037, -0.0036, -0.0035, -0.0037, -0.0036, -0.0037, 
+    -0.0032, -0.0032, -0.0028 ;
+
+ CNDC_QC = 4, 4, 4, 4, 4, 4, 4, 4, 4, 4 ;
+
+ PHASE = 119, 116, 116, 116, 116, 116, 116, 116, 116, 116 ;
```
Comment (vturpin) [inline examples/SEA076_20230906T0852_nrt.cdl:330]: What is the meaning of those values?

**Review by vturpin — COMMENTED** [review]
I don't see much to say. It may deserve a much closer look to make sure this is fully align with the format that will evolve anyway.
However, thanks for the great work Callum !

File: examples/SEA076_20230906T0852_nrt.cdl (hunk 1 above)
Comment (callumrollo) [reply examples/SEA076_20230906T0852_nrt.cdl:230]: That's what I did at first, but I found that it failed the CF1.8 checker https://cfchecker.ncas.ac.uk/

If I'm reading the CF docs correctly, they require strings of set length http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#_data_types

I'm happy to be corrected on this, but we need to make sure we're compliant with CF. Can we have plain string type with CF1.10?

File: examples/SEA076_20230906T0852_nrt.cdl (hunk 7 above)
Comment (callumrollo) [reply examples/SEA076_20230906T0852_nrt.cdl:331]: This is the nav state from the SeaExplorer that indicates e.g. when the glider is in communication with satellite. I've added a comment to this effect. I couldn't find a dive phase variable in the NERC OG vocabulary server. Is there one we should use? All I can find is GLCOMPH which has no indication of what values to use https://vocab.nerc.ac.uk/collection/OG1/current/GLCOMPH/

File: examples/SEA076_20230906T0852_nrt.cdl (hunk 3 above)
Comment (callumrollo) [reply examples/SEA076_20230906T0852_nrt.cdl:32]: I believe int8 is functionally the same as byte here I believe. No need to change the doc Victor

File: examples/SEA076_20230906T0852_nrt.cdl (hunk 1 above)
Comment (callumrollo) [reply examples/SEA076_20230906T0852_nrt.cdl]: After discussion this morning I have changed these to `string` type. We believe this is compliant with CF1.10

File: examples/SEA076_20230906T0852_nrt.cdl (hunk 7 above)
Comment (callumrollo) [reply examples/SEA076_20230906T0852_nrt.cdl]: I've removed PHASE until it is defined as part of OG1.0

**Review by vturpin — APPROVED** [review]

**Pull request merged**